dango/dex/src/core/order_matching.rs

```diff
@@ -92,3 +94,302 @@ where
         asks,
     })
 }
+
+pub fn match_market_orders<M, L>(
+    market_orders: &mut M,
+    limit_orders: &mut Peekable<L>,
+    market_order_direction: Direction,
+    maker_fee_rate: Udec128,
+    taker_fee_rate: Udec128,
+    current_block_height: u64,
+) -> anyhow::Result<Vec<FillingOutcome>>
+where
+    M: Iterator<Item = StdResult<(OrderId, MarketOrder)>>,
+    L: Iterator<Item = StdResult<((Udec128, OrderId), LimitOrder)>>,
+{
+    let mut maybe_market_order = market_orders.next().transpose()?;
+    let mut filling_outcomes = BTreeMap::<OrderId, FillingOutcome>::new();
+
+    // Limit order direction is assumed to be opposite to the market order direction
+    let limit_order_direction = match market_order_direction {
+        Direction::Bid => Direction::Ask,
+        Direction::Ask => Direction::Bid,
+    };
```

Comment:
I think we can implement `Neg` trait on `Direction`, and then here we can do:

```rust
let limit_order_direction = -market_order_direction;
```

Looks more neat this way!